### PR TITLE
Add sql grammar support for show flint index statement

### PR DESCRIPTION
**Pull request by seankao-az**
### Description
Add the sql antlr grammar support for `SHOW FLINT [INDEX|INDEXES]` statement, for [sql repo to download and use](https://github.com/opensearch-project/sql/blob/fcc4be3a6eea68b8c3ec5f649b53455a80655a35/spark/build.gradle#L17)

Here only the grammar files are changed. AST builder and implementation of the statement is prioritized to be done in the sql repo for response latency.

### Issues Resolved
* #265 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4 [hunk 1]
```diff
@@ -125,6 +126,14 @@ vacuumMaterializedViewStatement
     : VACUUM MATERIALIZED VIEW mvName=multipartIdentifier
     ;
 
+indexManagementStatement
+    : showFlintIndexStatement
+    ;
+
+showFlintIndexStatement
+    : SHOW FLINT (INDEX | INDEXES)
```
Comment (dai-chen) [inline flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:134]: Will this show Flint indexes in all catalogs or only current catalog?

Comment (seankao-az) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:134]: In current catalog.
For the sql plugin use case, it'll be
```
POST /_async_query
{
    "datasource": "glue_1",
    "lang": "sql",
    "query": "SHOW FLINT INDEX"
}
```

And show only the indexes in glue_1

Comment (seankao-az) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:134]: @dai-chen 
Do you think explicitly include catalogDb in the statement could be better?
`SHOW FLINT [INDEX|INDEXES] IN catalog[.databse]`

Comment (noCharger) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:134]: So we are offering this grammar only for SQL plugins?

Comment (seankao-az) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:134]: Prioritize for SQL plugin, yes. We want this in flint extension as well in the future.

Comment (noCharger) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:134]: > Prioritize for SQL plugin, yes. We want this in flint extension as well in the future.

In this case, can you add them in https://github.com/opensearch-project/sql/tree/fcc4be3a6eea68b8c3ec5f649b53455a80655a35/spark/src/main/antlr ?

Comment (seankao-az) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:134]: No.. spark repo and sql repo share the same grammar files. The grammar files in sql repo will be downloaded from spark repo when building

Comment (noCharger) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:134]: > No.. spark repo and sql repo share the same grammar files. The grammar files in sql repo will be downloaded from spark repo when building

we have previous PR to fix build errors where grammar missing in sql plugin https://github.com/opensearch-project/sql/pull/2476

Comment (seankao-az) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:134]: > > No.. spark repo and sql repo share the same grammar files. The grammar files in sql repo will be downloaded from spark repo when building
> 
> we have previous PR to fix build errors where grammar missing in sql plugin [opensearch-project/sql#2476](https://github.com/opensearch-project/sql/pull/2476)

I think that's just excluding the gen files from spotless formatter?
When I tried adding the grammar files in sql repo, they're still download from spark repo when building.iii

Comment (seankao-az) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4]: > Does this show Flint index in all catalogs or only current catalog?

> In current catalog. For the sql plugin use case, it'll be

Changing the statement for clarity.
`SHOW FLINT [INDEX|INDEXES] IN catalog[.database]`
Allowing user to show flint indexes in the specified catalog or database.

**Review by dai-chen — APPROVED** [review]

**Review by noCharger — APPROVED** [review]

**Comment by seankao-az** [comment]
Please help merge the PR, thanks!

**Pull request merged**